### PR TITLE
Add more core APIs

### PR DESCRIPTION
**Pull request by thc202**
Add core APIs to manage local proxies and rule configs, already present
in 2.8.0.

### CODE REVIEW

File: subprojects/zap-clientapi/src/main/java/org/zaproxy/clientapi/gen/RuleConfig.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        return api.callApi("ruleConfig", "action", "resetAllRuleConfigValues", null);
+    }
+
+    /** Set the specified rule configuration, which must already exist */
+    public ApiResponse setRuleConfigValue(String key, String value) throws ClientApiException {
+        Map<String, String> map = new HashMap<>();
+        map.put("key", key);
+        if (value != null) {
+            map.put("value", value);
+        }
```
Comment (kingthorin) [inline subprojects/zap-clientapi/src/main/java/org/zaproxy/clientapi/gen/RuleConfig.java]: Does null need to be handled? (Error response? Log?)

File: subprojects/zap-clientapi/src/main/java/org/zaproxy/clientapi/gen/LocalProxies.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        map.put("port", port);
+        if (behindnat != null) {
+            map.put("behindNat", behindnat);
+        }
+        if (alwaysdecodezip != null) {
+            map.put("alwaysDecodeZip", alwaysdecodezip);
+        }
+        if (removeunsupportedencodings != null) {
+            map.put("removeUnsupportedEncodings", removeunsupportedencodings);
+        }
```
Comment (kingthorin) [inline subprojects/zap-clientapi/src/main/java/org/zaproxy/clientapi/gen/LocalProxies.java]: Do any of these need to be handled as null? (Error response or log?)

File: subprojects/zap-clientapi/src/main/java/org/zaproxy/clientapi/gen/RuleConfig.java (hunk 1 above)
Comment (thc202) [reply subprojects/zap-clientapi/src/main/java/org/zaproxy/clientapi/gen/RuleConfig.java]: No, these parameters are optional, if null we don't send them at all.

File: subprojects/zap-clientapi/src/main/java/org/zaproxy/clientapi/gen/LocalProxies.java (hunk 2 above)
Comment (thc202) [reply subprojects/zap-clientapi/src/main/java/org/zaproxy/clientapi/gen/LocalProxies.java]: Same as above.

**Comment by psiinon** [comment]
Might ned to be rebased to pick up the openjdk build change?

**Comment by thc202** [comment]
Yes, done.

**Pull request merged**